samples/apps/basic/custom_endpoints/registry.h

```diff
@@ -0,0 +1,599 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+// CCF
+#include "ccf/app_interface.h"
+#include "ccf/common_auth_policies.h"
+#include "ccf/ds/hash.h"
+#include "ccf/http_query.h"
+#include "ccf/json_handler.h"
+#include "ccf/version.h"
+
+#include <charconv>
+#define FMT_HEADER_ONLY
+#include <fmt/format.h>
+
+// Custom Endpoints
+#include "ccf/bundle.h"
+#include "ccf/endpoint.h"
+#include "ccf/endpoints/authentication/js.h"
+#include "ccf/js/core/context.h"
+#include "ccf/js/core/wrapped_property_enum.h"
+#include "ccf/js/extensions/ccf/consensus.h"
+#include "ccf/js/extensions/ccf/converters.h"
+#include "ccf/js/extensions/ccf/crypto.h"
+#include "ccf/js/extensions/ccf/historical.h"
+#include "ccf/js/extensions/ccf/host.h"
+#include "ccf/js/extensions/ccf/kv.h"
+#include "ccf/js/extensions/ccf/request.h"
+#include "ccf/js/extensions/ccf/rpc.h"
+#include "ccf/js/extensions/console.h"
+#include "ccf/js/extensions/math/random.h"
+#include "ccf/js/modules.h"
+#include "ccf/service/tables/modules.h"
+#include "endpoint.h"
+#include "js/interpreter_cache_interface.h"
+
+using namespace nlohmann;
+
+namespace basicapp
+{
+  class CustomJSEndpointRegistry : public ccf::UserEndpointRegistry
```

Comment:
Can you add a docstring to this, outlining the broad goal and listing the known TBDs (vs the js_generic endpoints)? We should have a central place to say "doesn't handle templated endpoints, doesn't utilise the bytecode cache" for the first dev release.

Reply:
Yes, very good point.